multisig-wallet/src/main.sw

```diff
@@ -4,29 +4,34 @@ use std::{
     address::Address,
     assert::require,
     b512::B512,
-    context::call_frames::contract_id,
+    chain::auth::Sender,
+    context::{call_frames::contract_id, this_balance},
     contract_id::ContractId,
     ecr::{EcRecoverError, ec_recover_address},
     hash::sha256,
     logging::log,
     result::*,
     revert::revert,
-    storage::{get, store}
+    storage::{get, store},
+    token::{force_transfer, transfer_to_output}
 };
 
 use core::num::*;
 
 abi MultiSignatureWallet {
     fn constructor(owner1: Address, owner2: Address, owner1_weight: u64, owner2_weight: u64, threshold: u64) -> bool;
-    fn execute_transaction(to: ContractId, value: u64, data: b256, signature1: B512, signature2: B512) -> bool;
+    fn execute_transaction(to: Sender, value: u64, data: b256, signature1: B512, signature2: B512) -> bool;
```

Comment:
Maybe I'm missing something, but get_transaction_hash and _get_transaction_hash both return a b256. Why do execute_transaction and transfer take b512 signatures?

Reply:
The hashing functions take some parameters and produce a hash which is meant to be signed with the signature of the owner(s). More information regarding the signature structure can be found in [EIP-2098: Compact Signature Representation](https://eips.ethereum.org/EIPS/eip-2098).

Reply:
Oh, and as for the "why", it is to make sure that when the public key is recovered from the signature that key is regarded as an owner in the multisig. This is to prevent random people from interacting with the contract.